api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -180,36 +170,45 @@ export class ProjectAclService implements ProjectAccessControl {
 
   async updateUserInProject(
     projectId: string,
-    updateUserInProjectDto: UserRoleInProjectDto,
+    userAndRoleToChange: UserRoleInProjectDto,
     loggedUserId: string,
   ): Promise<Either<Permit, void>> {
-    const { userId, roleName } = updateUserInProjectDto;
+    const { userId, roleName } = userAndRoleToChange;
     if (!(await this.isOwner(loggedUserId, projectId))) {
       return left(false);
     }
-
+    assertDefined(roleName);
     const apiDbConnection = getConnection(DbConnections.default);
     const apiQueryRunner = apiDbConnection.createQueryRunner();
 
     await apiQueryRunner.connect();
     await apiQueryRunner.startTransaction();
 
     try {
-      const existingUserInProject = await this.roles.findOne({
-        where: {
-          projectId,
-          userId,
+      const existingUserInScenario = await apiQueryRunner.manager.findOne(
+        UsersProjectsApiEntity,
+        undefined,
+        {
+          where: {
+            projectId,
+            userId,
+          },
         },
-      });
+      );
 
-      if (!existingUserInProject) {
-        await this.roles.save({
-          projectId,
-          userId,
-          roleName,
-        });
+      if (!existingUserInScenario) {
+        const userRoleToSave = new UsersProjectsApiEntity();
+        userRoleToSave.projectId = projectId;
+        userRoleToSave.userId = userId;
+        userRoleToSave.roleName = roleName;
+
+        await apiQueryRunner.manager.save(userRoleToSave);
       } else {
-        await this.roles.update({ projectId, userId }, { roleName });
+        await apiQueryRunner.manager.update(
+          UsersProjectsApiEntity,
+          { projectId, userId },
+          { roleName },
+        );
       }
     } catch (err) {
       await apiQueryRunner.rollbackTransaction();
```

Comment:
@rubvalave has this been addressed?

Reply:
Yes, this can be closed (Though I have to do the same for project-acl.service, I will add it to this PR as well because why not a few more lines of code)